### ISSUE TITLE
order of data  with horizontal  barchart

### ISSUE THREAD

**Issue by wykhuh** [britecharts/britecharts#662]
## Expected Behavior
When using `isHorizontal(true)` for a bar chart, I expect the first item in the dataset to be the first bar at the top, and the last item in the dataset to be the last bar at the bottom.

## Current Behavior
Currently, when using `isHorizontal(true)` for a bar chart,  the first item in the dataset  is the last bar at the bottom, and the last item in the dataset is the  first bar at the top.

## Screenshots:
 I turned the letter bar chart demo into a horizontal chart on codepen: https://codepen.io/wykhuh/pen/VGGmXQ?editors=0010
<img width="302" alt="screen shot 2018-09-15 at 9 47 22 pm" src="https://user-images.githubusercontent.com/6968611/45593040-ba018d00-b931-11e8-8b2e-c0859e3357f6.png">

## Context
 In order to use `.isHorizontal(true)`, I currently have to manually reverse the order of the dataset in order for the first item of the dataset to be the first bar in the horizontal chart. It would be preferable if britechart would  put the first item at the top, or offer the option of putting the first item at the top or the  bottom.

## Your Environment
* Version used: 2.9.6
* Browser Name and version: Chrome
* Operating System and version:  Mac
* codepen: https://codepen.io/wykhuh/pen/VGGmXQ?editors=0010

**Comment by Golodhros** [comment]
Hi @wykhuh thanks for filling an issue!

Good catch! Thanks for reporting this bug, it should work as you describe.

I am not sure if we can fix it right away or wait for the next major release, as this change would modify the aspect of charts currently running. We'll discuss it within the team and let you know.

Thanks again!

**Comment by ferreiro-eb** [comment]
Hi there! I was checking the code and I come up with one way for implementing this. It relies on a boolean flag, for example: `isReverse`, to offer backwards compatibility. Having this flag, we can check later in the code if the user wants to reverse and then reverse the bars list.

### What to modify?
In `drawBars` method, before check if isHorizontal(),  we can check if isReverse(), if so we can update bars to bars.reverse().

```
function drawBars() {
    let bars;

    if (isAnimated) {
        bars = svg.select('.chart-group').selectAll('.bar')
            .data(dataZeroed);

        if (isReverse) {
            bars = bars.reverse();
        }

        if (isHorizontal) {
            drawHorizontalBars(bars);
        } else {
            drawVerticalBars(bars);
        }

   ...
}
```

I also saw that we had a function called `sortData`, but not sure if works in this case.

@miglesiasEB  What do you think about that propsal? Does it make sense to move forward with it or do you have any other ideas?

**Comment by Golodhros** [comment]
Thanks a lot @ferreiro-eb!

The problem seems to be in this line: https://github.com/Golodhros/britecharts/blob/master/src/charts/bar.js#L443 as it seems we are starting to draw the bars at the bottom (D3 0,0 point is at the top left). We should start drawing in y = 0 instead, and take it from there...

I think we would be OK by fixing this directly without worrying about backward compatibility.

**Comment by mmainzer** [comment]
Has there been update to this issue? How can I reverse the order my data is being displayed in the bar without having to manually reorder the data itself?

**Comment by Golodhros** [comment]
Hi @mmainzer,

I think for now you can call `.reverse()` in your data before feeding it to your bar chart.

**Comment by Golodhros** [comment]
Fixed in #921
